### ISSUE TITLE
Resizing List causes visible items to refresh instead of resize

### ISSUE THREAD

**Issue by kendellfab** [fyne-io/fyne#4080]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When an image is on one side of a split view, scrubbing the split view back and forth has a large delay as the image is being updated. This was working well on the v2.3.5 release, but appeared in the develop branch.

### How to reproduce

1. Create a window with a split
2. One side of the split add a basic widget (i.e. a label)
3. On the other side add an image (it might be size dependent on how big the impact is to performance)
4. Scrub the split view left and right

### Screenshots

https://github.com/fyne-io/fyne/assets/1508163/ba7bcd62-793d-4b93-a3b0-0c237de995c3

### Example code

package main

import (
	"encoding/base64"
	"log"

	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/canvas"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
)

func main() {
	regression := app.New()
	window := regression.NewWindow("Regression")
	window.Resize(fyne.NewSize(1200, 700))
	
	leftLbl := widget.NewLabel("Left Side")
	
        // I can't put a large enough image into this text editor on github for an example, but I'm seeing this with images that are 600x400 and up, use one of those
	bs, err := base64.StdEncoding.DecodeString(imageData)
	if err != nil {
		log.Fatalln("error loading image data", err)
	}
	
	img := canvas.NewImageFromResource(fyne.NewStaticResource("example", bs))
	
	split := container.NewHSplit(leftLbl, img)
	split.SetOffset(0.25)
	
	window.SetContent(split)
	window.ShowAndRun()
}

const (
imageData = ``
)

### Fyne version

develop

### Go compiler version

1.20.6

### Operating system and version

Arch Linux

**Comment by andydotxyz** [comment]
Ah, maybe we have refreshed too deeply on a resize following changes.

Does this change sort things for you? (asking in case I can't get a suitable image locally...)

```diff
diff --git a/canvas/image.go b/canvas/image.go
index e22381fbd..ae0cf3fa5 100644
--- a/canvas/image.go
+++ b/canvas/image.go
@@ -180,7 +180,7 @@ func (i *Image) Resize(s fyne.Size) {
        }
 
        i.baseObject.Resize(s)
-       i.Refresh()
+       Refresh(i)
 }
 
 // NewImageFromFile creates a new image from a local file.
```

**Comment by kendellfab** [comment]
I tried this update in the one off example that I posted above. It helped there. So I tried it in the app shown above, and it didn't hep as much as I'd expect. So I played around with the app, and noticed that there was that sluggishness even without showing the big image. 

One thing I noticed, in the video attached, is the highlights for the list rows seem to flicker and cause some slow down as well.

https://github.com/fyne-io/fyne/assets/1508163/4c47e4e7-4a37-44e5-8af4-b92764f5634b

**Comment by andydotxyz** [comment]
> One thing I noticed, in the video attached, is the highlights for the list rows seem to flicker and cause some slow down as well.

It's just a guess but looks like you have some pretty custom list items that have perhaps overridden the Tapped or Hover handling and it's going round in circles? Additionally it seems that resizing your list items is slow - perhaps something in your custom item widget could be impacting that?

**Comment by kendellfab** [comment]
They are custom widgets, but they do not override any tapping or hovering methods. I can look into the code for the layout, but I didn't notice this issue until the latest development branch, which to me would indicate another performance issue in fyne somewhere.

**Comment by andydotxyz** [comment]
The focus handling has changed in collections with 2.4 which could be why you didn't get this before...

**Comment by kendellfab** [comment]
How then, would one go about addressing this change in performance in their application? As the update took a usable app and made it impossibly sluggish.

**Comment by andydotxyz** [comment]
> How then, would one go about addressing this change in performance in their application? As the update took a usable app and made it impossibly sluggish.

Sorry I was referring to the flicker of select / focus on the list.

As for performance I don't know - can you help us understand what is happening with a CPU profile report? None of our test apps are demonstrating this performance hit.

**Comment by kendellfab** [comment]
I've run pprof.StartCPUProfile with the latest version of fyne and the development version. How would you like me to send those profiles?

**Comment by andydotxyz** [comment]
I'm sorry I missed this message. Can you attach an SVG output of the report perhaps? I think GitHub is happy with that file type.

**Comment by kendellfab** [comment]
**Latest Build**

![weatherlight-fyne-v2 3 5](https://github.com/fyne-io/fyne/assets/1508163/633acf23-9388-4218-b983-b02693bb475e)

**Develop**


![weatherlight-fyne-develop](https://github.com/fyne-io/fyne/assets/1508163/7975ab42-da83-4354-ba87-aad7f3509ef5)

**Comment by kendellfab** [comment]
Yesterday I had some time to test this some more, and I tried it on my M2 Macbook air, and didn't notice any of the sluggishness. I could try it on windows if needed as well.

**Comment by andydotxyz** [comment]
It would be good to know how things are with latest develop, most of the improvements are now landed.

**Comment by andydotxyz** [comment]
From the CPU profiles it does look like it should be fixed from the change I added - it was continually re-loading the jpeg which is what I resolved.
Unless something further up the UI is calling `Refresh` on the image each time the `Split` is moved?

**Comment by kendellfab** [comment]
I have been digging into this a bit more.  As a quick aside, I have lists that are the first child widgets of two splits.

I took out all of my fancy layouts, and left just the images and the names. I then added loggers in my code that updates the images. And I've found that yes, the images are getting set a lot. But the trick is, this is coming from the UpdateTemplate function that is attached to the list. It appears as though, scrubbing the split is causing the UpdateTemplate functions to be called many times on each list that is being displayed.

Why then would the UpdateTemplate function be called on a split scrub?
